sourcegen-plugin-annotations/src/main/java/io/micronaut/sourcegen/annotations/GenerateGradlePlugin.java

```diff
@@ -106,6 +109,17 @@ enum Type {
          */
         String extensionMethodName() default "";
 
+        /**
+         * @return Whether the task is cacheable
+         */
+        boolean cacheable() default true;
+
+        /**
+         * @return Path sensitivity to use for file parameters must be one of
+         * {@code org.gradle.api.tasks.PathSensitivity} values: ABSOLUTE, RELATIVE, NAME_ONLY, NONE.
+         */
+        String pathSensitivity() default "ABSOLUTE";
```

Comment:
should this be an enum?

Reply:
This should actually be per property, not a single entry for all paths.